### ISSUE TITLE
IsADirectoryError: [Errno 21] Is a directory:

### ISSUE THREAD

**Issue by omerlh** [semgrep/semgrep#781]
**Describe the bug**

When trying to run the latest version from develop (commit ad6260f79fc134292dd57bbb9cc395d7d9da1dc6), I'm getting the following unclear error:

```
semgrep -f ../../registry/.semgrep.yml ../../<> --exclude-dir=node_modules 
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/omerlh/dev/semgrep/semgrep/semgrep/core_runner.py", line 38, in get_re_matches
    contents = path.read_text()
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pathlib.py", line 1216, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pathlib.py", line 1203, in open
    opener=self._opener)
IsADirectoryError: [Errno 21] Is a directory: '<>'
"""
```

**Environment**
If not using semgrep.live: are you running off docker, an official binary, a local build?

**Comment by underyx** [comment]
[As discussed on Twitter](https://twitter.com/underyx/status/1262704255179395074), our immediate advice just to unblock you, is that you try one of the binaries we built, as `develop` is considered unstable.

@mschwager could you try to reproduce this to ship an actual fix?

**Comment by omerlh** [comment]
I know, but I was looking for the regex support which look like is not available on the latest released version... this is why I tried develop branch

**Comment by underyx** [comment]
Ah, gotcha. That feature's very fresh and might not be ready for prime time. We're planning to run a QA pass and release a new binary today (US west coast time), so you can look forward to that.

But if you're impatient: I think I might've tracked down the source of the error you saw, and you can give this commit a go: https://github.com/returntocorp/semgrep/commit/5db300583a9d4036a3cb5951f7a79f4b231ef121

**Comment by omerlh** [comment]
This solved but now I'm having a new issue:
```
Traceback (most recent call last):
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/omerlh/dev/semgrep/semgrep/semgrep/core_runner.py", line 38, in get_re_matches
    contents = path.read_text()
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pathlib.py", line 1217, in read_text
    return f.read()
  File "/usr/local/opt/python/bin/../Frameworks/Python.framework/Versions/3.7/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

**Comment by underyx** [comment]
Ah, I see, we've always filtered to files with language specific file extensions before the regex matcher was added, and seems like we overlooked that now binary files might end up being scanned. This commit adds a test case and fixes this: https://github.com/returntocorp/semgrep/pull/783/commits/c9e9260b02f346027f55ab40f4389c12b7376268

Sorry about all these hoops! We'll be doing plenty of testing before the stable release later today, so feel free to just wait until we're done with that.

**Comment by omerlh** [comment]
Awesome! Does that means the regex will be part of this release?

**Comment by underyx** [comment]
That's right!

**Comment by underyx** [comment]
@omerlh just wanted to let you know, we added a QA suite for testing regex on real world repos, and fixed another issue regarding metadata output on regex pattern matches. Sadly, we ran out of time during yesterday, so the release including `pattern-regex` had to be pushed back till today :(

**Comment by omerlh** [comment]
No worries :) happy to hear you spend a lot of time on quality :)

**Comment by underyx** [comment]
Hey again @omerlh, got some good news! 0.8.0 is now out with stable regex support. You can upgrade now with `brew install returntocorp/semgrep/semgrep` (there's a bit of lag before the official Homebrew formula is updated).
